sentence_transformers/SentenceTransformer.py

```diff
@@ -370,8 +376,9 @@ def encode(
                         row = {name: out_features[name][sent_idx] for name in out_features}
                         embeddings.append(row)
```

Comment:
Should truncation also be done when `output_value=None`? We'd only truncate the `out_features["sentence_embedding"]`

Reply:
That does seem like the behaviour that people would expect, yeah.

Reply:
Makes sense. Though this makes me wonder if it's more consistent to truncate `out_features["sentence_embedding"]` even earlier—after the forward pass of the pooler module. Maybe we'd have `Truncate` be its own torch module which gets appended to `self.modules`. That'd make truncation apply to training as well. Though that seems like a non-existent use case since Matryoshka training gets you more w/o any real computational cost

I'm not really for a `Truncate` module b/c it feels like too much to connect truncation to everything about a model, including how it gets trained and saved. I see truncation more as a little inference trick. Curious if you have thoughts on this

Reply:
Fixed and tested in 8125182206bcde523da519d09e92f70ec9813312

Reply:
Personally, I think the module-based design can be a bit complicated, so I prefer this solution I think.